ui/app/common/concept-set/mappers/observationMapper.js

```diff
@@ -4,26 +4,34 @@ Bahmni.ConceptSet.ObservationMapper = function () {
     var conceptMapper = new Bahmni.Common.Domain.ConceptMapper();
     var self = this;
     // TODO : Shouldn't this be in clinical module. Don't see a reason for this to be in concept-set code - Shruthi
-    this.getObservationsForView = function (observations, conceptSetConfig) {
-        return internalMapForDisplay(observations, conceptSetConfig);
+    this.getObservationsForView = function (observations, conceptSetConfig, $translate) {
+        return internalMapForDisplay(observations, conceptSetConfig, $translate);
     };
-
-    var internalMapForDisplay = function (observations, conceptSetConfig) {
+    var internalMapForDisplay = function (
+      observations,
+      conceptSetConfig,
+      $translate
+    ) {
         var observationsForDisplay = [];
         _.forEach(observations, function (savedObs) {
             if (savedObs.concept.conceptClass && (savedObs.concept.conceptClass === Bahmni.Common.Constants.conceptDetailsClassName || savedObs.concept.conceptClass.name === Bahmni.Common.Constants.conceptDetailsClassName)) {
-                var observationNode = new Bahmni.ConceptSet.ObservationNode(savedObs, savedObs, [], savedObs.concept);
-                var obsToDisplay = createObservationForDisplay(observationNode, observationNode.primaryObs.concept);
-                if (obsToDisplay) {
-                    observationsForDisplay.push(obsToDisplay);
+                if (isBahmniLite(observations, $translate)) {
```

Comment:
I am not sure why this condition exists? `isBahmniLite()`.

Ideally there should not be any code in Bahmni, that checks for whether Bahmni is running in LITE / STANDARD version. There should be only one code, which is being used in Bahmni UI, and configuration decides what gets shown on screen. 

Please clarify, maybe the variable name is misleading. thanks.

Reply:
https://github.com/Bahmni/openmrs-module-bahmniapps/blob/1331939dd7e486dac6bb4bc897e70b943685b870/ui/app/common/obs/models/observation.js#L65

The reason we went for this approach where we distinguish between LITE / STANDARD version is due to the fact that the response from Bahmni core varies for both the platforms. We checked how this was previously handled and decided to go ahead with the same. Above is an example of the same where in response to the platform  determines what action is to be taken. The variable name as you suggested could have been done better. Apart from the variable name, the approach is similar to what has been already done in `ui/app/common/obs/models/observation.js#L65`.

Reply:
> the response from Bahmni core varies for both the platforms
Bahmni LITE runs with a different config folder than Bahmni STANDARD. And implementation partners that adopt Bahmni will modify these configurations to suit their needs. So, in either scenario, responses change depending on configuration (and not if they are LITE/Standard). 

Both Bahmni LITE and STANDARD run on the same OpenMRS+Bahmni codebase.

If we are writing this to ensure backward compatible behavior, then maybe we need to decide what is old behaviour and new behavior, and discuss what methodName/variable name will make sense in code so that it is clearer. Ideally, if this is about translation messages -- we need to ensure old config and new config, both show up as same on screen. 

Happy to discuss further. I might be misunderstanding the platform behaviors/responses. thanks.

Reply:
Sure @gsluthra, we'll try to address this issue as soon as possible. We will connect with you on the same.